### ISSUE TITLE
fix email flow

### ISSUE THREAD

**Issue by russhaun** [russhaun/artillery#52]
email func is not working as intended(i think). in honeypot.py in main tcp socket class the func warn_the_good_guys() has no check to see if enabled. this is remedied elsewhere to prevent sending related to previous issues but even with the check u still get an alert when u start artillery. suggest reworking check to prevent message and correct flow to what u expect also this func resides inside a class so it is created every time for each port similar to banip func this should be placed outside class on a timer of sorts resulting in smaller emails and not so many

**Comment by russhaun** [comment]
unrelated to flow. listen_udp and listen_tcp servers were bypassing checks and making a call to mail directly if ports were not open. changed to warn_the_good_guys added additional logic inside warn_the_good_guys to act accordingly based on settings.

**Comment by russhaun** [comment]
udp and tcp server related issues were fixed in latest 1.0.5-beta. main issue still exists

**Comment by russhaun** [comment]
moved functions from core.py related to email to email_handler.py to put similar functions together. sending email on timer works as expected with some minor formatting issues. sending right away still has issues. suggest if using email to use timer to stop spamming. working on solution now. also if 2fa is enabled on acct config option availible to set on windows

**Comment by russhaun** [comment]
I have created a new email class to deal with duplicate emails being sent. This new class when run will monitor a file and when this file is modified it will compile and send 1 ill say again 1 email with all relevant info attached as a zipfile to the address specified. pretty sure this "fixes" the issue with duplicate emails. I have also added methods to allow changing the subject line and alertlog used for emails to allow being used for other things other then honeypot attacks.
